lib/osf-components/addon/components/addons-service/user-addons-manager/component.ts

```diff
@@ -137,13 +177,46 @@ export default class UserAddonManagerComponent extends Component<Args> {
         }
     }
 
+    @task
+    @waitFor
+    async getAuthorizedCitationAccounts() {
+        const { userReference } = this;
+        const mappedObject = this.filterTypeMapper[FilterTypes.CITATION_MANAGER];
+        if (userReference) {
```

Comment:
Especially for the user-addons-manager, which would only be on the user settings page, shouldn't we always have a userReference here? Or is it a timing issue while the API request is being made?

Reply:
There _should_ be a userReference, as it is the first thing done when we initialize this component, so I don't think there should be a timing issue. I've defined userReference as an optional property on this (`userReference?: UserReferenceModel`), just to be safe, and these `if` statements are a consequence of that typing. I think I'm leaning towards doing `userReference!:` though, as this component doesn't make much sense without it, as you said